hconf.c

```diff
@@ -149,6 +156,77 @@ hconf_input_mode_handler(SYSCTL_HANDLER_ARGS)
         return (0);
 }
 
+static int
+hconf_get_switch(struct hconf_softc *sc, int swtype, u_int *mask)
+{
+	uint8_t *fbuf;
+	int error;
+
+	if (sc->switch_rlen[swtype] <= 1)
+		return (ENXIO);
+
+	fbuf = malloc(sc->switch_rlen[swtype], M_TEMP, M_WAITOK | M_ZERO);
+	sx_xlock(&sc->lock);
+
+	error = hid_get_report(sc->dev, fbuf, sc->switch_rlen[swtype], NULL,
+	    HID_FEATURE_REPORT, sc->switch_rid[swtype]);
+	if (error == 0) {
+		*mask = hid_get_data_unsigned(fbuf + 1,
+		    sc->switch_rlen[swtype] - 1, &sc->switch_loc[swtype]);
+	}
+
+	sx_unlock(&sc->lock);
+	free(fbuf, M_TEMP);
+	return (error);
+}
+
+static int
+hconf_set_switch(struct hconf_softc *sc, int swtype, u_int mask)
```

Comment:
hconf_set_switch() should be called from device_resume handler to as suspend/resume cucle often resets touchpad settings.